frontend/src/components/widgets/inputs/ColorInput.vue

```diff
@@ -98,16 +95,17 @@ export default defineComponent({
 		};
 	},
 	computed: {
-		color() {
-			if (!this.value) return { r: 0, g: 0, b: 0, a: 1 };
-
-			const r = parseInt(this.value.slice(0, 2), 16);
-			const g = parseInt(this.value.slice(2, 4), 16);
-			const b = parseInt(this.value.slice(4, 6), 16);
-			const a = parseInt(this.value.slice(6, 8), 16);
-			return { r, g, b, a: a / 255 };
+		color(): Color {
+			if (!this.value) return new Color(0, 0, 0, 1);
+
+			const r = parseInt(this.value.slice(0, 2), 16) / 255;
+			const g = parseInt(this.value.slice(2, 4), 16) / 255;
+			const b = parseInt(this.value.slice(4, 6), 16) / 255;
+			const a = parseInt(this.value.slice(6, 8), 16) / 255;
+
```

Comment:
don't know if this is cleaner but you could do

```ts
const [r, g, b, a] = this.value.match(/.{1,2}/g).map(byte => parseInt(byte, 16))
```
the match(/.{1,2}/g) chunks the string into pairs https://stackoverflow.com/questions/7033639/split-large-string-in-n-size-chunks-in-javascript

You also have to make the if statement above check for non conforming lengths

Reply:
I think the clarity of the current solution is worth the three lines, but I appreciate the cleverness and that's a new thing I just learned about regex, thanks. I'll add the length check guard to the if statement as you suggested.